ariadne_codegen/plugins/shorter_results.py

```diff
@@ -0,0 +1,179 @@
+        # statement?
+        return_stmt = method_def.body[-1]
+        if not isinstance(return_stmt, ast.Return):
+            return super().generate_client_method(method_def)
+
+        # Get the originally returned class so we can look at the class
+        # definition to see how many fields it has.
+        if not isinstance(method_def.returns, ast.Name):
+            return super().generate_client_method(method_def)
+
+        return_class = method_def.returns.id
+        if return_class not in self.class_dict:
+            return super().generate_client_method(method_def)
+
+        return_class_ast = self.class_dict[return_class]
+
+        # If we're not a single method data class move on!
+        if len(return_class_ast.body) != 1:
+            return super().generate_client_method(method_def)
+
+        single_field = return_class_ast.body[0]
+
+        # Get the field class name, which might be an optional and if so
+        # represented as an `ast.Subscript`.
+        # We need to update the proper return type to be optional as well so we
+        # store that. The ast in the annotation isn't an evaluated node so we
+        # can't return that as is.
+        if isinstance(single_field.annotation, ast.Name):
+            is_optional = False
+            single_field_class = single_field.annotation.id
+        elif isinstance(single_field.annotation, ast.Subscript):
+            is_optional = True
+            single_field_class = single_field.annotation.slice.id
+        else:
+            return super().generate_client_method(method_def)
+
+        single_field_name = single_field.target.id
+
+        # We only want to expand for classes we created. Some queries will
+        # return a type that has a single field which is of `typing.Any` or
+        # other things not generated this iteration so ignore those.
+        # TODO: Can we ensure this in a better way?
+        try:
+            single_field_class = ast.literal_eval(single_field_class)
+        except ValueError:
+            # Not a literal
+            pass
+
+        if single_field_class not in self.class_dict:
+            return super().generate_client_method(method_def)
+
+        # Update the return type to be the inner field and ensure we reference
+        # the single field before returning.
+        if is_optional:
+            method_def.returns = ast.Subscript(
+                value=ast.Name(id="Optional"), slice=ast.Name(id=single_field_class)
+            )
+        else:
+            method_def.returns = ast.Name(id=single_field_class)
+
+        method_def.body[-1] = ast.Attribute(
+            value=return_stmt,
+            attr=single_field_name,
+        )
+
+        # After we change the type we also need to import it from the client so
+        # add the extra import as needed.
+        if method_def.name not in self.extended_imports:
+            self.extended_imports[method_def.name] = set()
```

Comment:
This will break if the method name isn't the same as the filename but I didn't find any other way to figure out which file would need to be updated. Are there better alternatives to use as a key to ensure we can pick this up in `generate_client_module`?